src/service/resources-impl.js

```diff
@@ -70,6 +70,315 @@ let MarginChangeDef;
+   * window, and (2) meet the filterFn given.
+   * @param {!Window} hostWin
+   * @param {function(!Resource):boolean} filterFn
+   * @return {!Promise<!Array<!Resource>>}
+   */
+  getMeasuredResources(hostWin, filterFn) {}
+
+  /**
+   * Returns a subset of resources which are (1) belong to the specified host
+   * window, and (2) positioned in the specified rect.
+   * @param {!Window} hostWin
+   * @param {!../layout-rect.LayoutRectDef} rect
+   * @param {boolean=} opt_isInPrerender signifies if we are in prerender mode.
+   * @return {!Promise<!Array<!Resource>>}
+   */
+  getResourcesInRect(hostWin, rect, opt_isInPrerender) {}
+
+  /**
+   * Returns the {@link Resource} instance corresponding to the specified AMP
+   * Element. If no Resource is found, the exception is thrown.
+   * @param {!AmpElement} element
+   * @return {!Resource}
+   */
+  getResourceForElement(element) {}
+
+  /**
+   * Returns the {@link Resource} instance corresponding to the specified AMP
+   * Element. Returns null if no resource is found.
+   * @param {!AmpElement} element
+   * @return {?Resource}
+   */
+  getResourceForElementOptional(element) {}
+
+  /**
+   * Returns the direction the user last scrolled.
+   *  - -1 for scrolling up
+   *  - 1 for scrolling down
+   *  - Defaults to 1
+   * TODO(lannka): this method should not belong to resources.
+   * @return {number}
+   */
+  getScrollDirection() {}
+
+  /**
+   * Signals that an element has been added to the DOM. Resources manager
+   * will start tracking it from this point on.
+   * @param {!AmpElement} element
+   */
+  add(element) {}
+
+  /**
+   * Signals that an element has been upgraded to the DOM. Resources manager
+   * will perform build and enable layout/viewport signals for this element.
+   * @param {!AmpElement} element
+   */
+  upgraded(element) {}
+
+  /**
+   * Signals that an element has been removed to the DOM. Resources manager
+   * will stop tracking it from this point on.
+   * @param {!AmpElement} element
+   */
+  remove(element) {}
+
+  /**
+   * Removes all resources belonging to the specified child window.
+   * TODO(lannka): this method should not belong to resources.
+   * @param {!Window} childWin
+   */
+  removeForChildWindow(childWin) {}
+
+  /**
+   * Schedules the work pass at the latest with the specified delay.
+   * @param {number=} opt_delay
+   * @param {boolean=} opt_relayoutAll
+   * @return {boolean}
+   */
+  schedulePass(opt_delay, opt_relayoutAll) {}
+
+  /**
+   * Registers a callback to be called when the next pass happens.
+   * @param {function()} callback
+   */
+  onNextPass(callback) {}
+
+  /**
+   * Called when main AMP binary is fully initialized.
+   * May never be called in Shadow Mode.
+   */
+  ampInitComplete() {}
+
+  /**
+   * Requires the layout of the specified element or top-level sub-elements
+   * within.
+   * @param {!Element} element
+   * @param {number=} opt_parentPriority
+   * @return {!Promise}
+   * @restricted
```

Comment:
@jridgewell this `restricted` annotation is copied over. but do you think it's needed here? This method should be public, it's used outside of this file.

Reply:
No, I don't think it's needed.

Reply:
removed